test/test_s3mock.py

```diff
@@ -75,18 +128,157 @@ def test_s3_mock_yml_mod_loc(self, file, powerloader_binary):
+            + "/"
+            + path_to_name(file["xtensor_path"])
+        )
         generate_s3_download_yml(file, file["s3_mock_server"], filename)
         download_s3_file(powerloader_binary, file, plain_http=True)
         Path(file["tmp_yml"]).unlink()
 
-        for fp in get_files(file):
+        for fp in get_files(file, expect=">zero"):
             assert calculate_sha256(fp) == checksums[str(path_to_name(fp))]
+
+    @skip_aws_credentials
+    def test_zchunk_basic(self, file, powerloader_binary):
+        # Download the expected file
+        name = file["lorem_zck_file"].name
+        localpath = file["tmp_path"] / name
+        assert not localpath.exists()
+
+        get_zchunk_s3(
+            file,
+            filepath=file["lorem_zck_file"],
+            powerloader_binary=powerloader_binary,
+            localpath=localpath,
+        )
+        """
+        headers = get_prev_headers(mock_server_working, 2)
+        assert headers[0]["Range"] == "bytes=0-256"
+        assert headers[1]["Range"] == "bytes=257-4822"
+        """
+        assert localpath.exists()
+        assert localpath.stat().st_size == 4823
+        assert not Path(str(localpath) + ".pdpart").exists()
+
+        """
+        clear_prev_headers(mock_server_working)
+        """
+        get_zchunk_s3(
+            file,
+            filepath=file["lorem_zck_file"],
+            powerloader_binary=powerloader_binary,
+            localpath=localpath,
+        )
+
+        """
+        headers = get_prev_headers(mock_server_working, 100)
+        assert headers is None
+        """
+        assert localpath.exists()
+        assert localpath.stat().st_size == 4823
+        assert not Path(str(localpath) + ".pdpart").exists()
+
+        new_name = Path("lorem.txt.x3.zck")
+        new_filepath = file["lorem_zck"] / new_name
+
+        """
+        clear_prev_headers(mock_server_working)
+        """
+
+        get_zchunk_s3(
+            file,
+            filepath=file["lorem_zck_file_x3"],
+            powerloader_binary=powerloader_binary,
+            localpath=localpath,
+        )
+
+        """
+        headers = get_prev_headers(mock_server_working, 100)
+        assert len(headers) == 2
+
+        # lead
+        assert headers[0]["Range"] == "bytes=0-88"
+        # header
+        assert headers[1]["Range"] == "bytes=0-257"
+        range_start = int(headers[2]["Range"][len("bytes=") :].split("-")[0])
+        assert range_start > 4000
+        """
+
+        # TODO: Specify output path, so that test/tmp/lorem.txt.zck is modified
+        # TODO: Download with CLI params rather than YML file, so that the headers can be checked easily
+        if False:
+            assert localpath.stat().st_size == new_filepath.stat().st_size
+            assert calculate_sha256(new_filepath) == calculate_sha256(str(localpath))
+            assert localpath.exists()
+            assert not Path(str(localpath) + ".pdpart").exists()
+            localpath.unlink()
```

Comment:
Perhaps this can also be removed and I can just use the workaround permanently..